python/cudf/cudf/core/series.py

```diff
@@ -3370,7 +3370,12 @@ def _describe_categorical(self):
             data = [self.count(), self.unique().size]
 
             if data[1] > 0:
```

Comment:
I was confused the first couple times I read through this function and I think there is a worthwhile refactor to be done. It's using two lists (`index` and `data`) to track information that acts like keys and values. For clarity, could we refactor this to use a dictionary like `{"count": count, "unique": unique, "top": top, "freq": freq}` and return a Series constructed from that?

The [pandas implementation](https://github.com/pandas-dev/pandas/blob/945c9ed766a61c7d2c0a7cbb251b6edebf9cb7d5/pandas/core/describe.py#L248-L277) uses two lists as well, but it constructs the list of data all at once. The cudf implementation is especially confusing because it builds by concatenation.

Reply:
Great suggestion.

Reply:
The last thing I was looking into yesterday was this xfailed test: https://github.com/rapidsai/cudf/blob/024003ca444f9d1a8374a1133337419f22cc880a/python/cudf/cudf/tests/test_series.py#L405. I'm gonna take a second dive today to see if it's a simple fix.

Reply:
I wanted to add an FYI here that my suggestion to use a `dict` relies on having ordered dictionaries. Dictionaries are insertion-ordered in Python 3.7+. That may explain why pandas did not use this approach when its function was originally implemented.